app/Http/Controllers/TradeOfferController.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\TradeStoreRequest;
+use App\Http\Requests\TradeUpdateRequest;
+use App\Models\Character;
+use App\Models\Item;
+use App\Models\Trade;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+use function PHPUnit\Framework\isNull;
+
+class TradeOfferController extends Controller
+{
+    /**
+     * Retrieve character's items that are valid to trade
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function create(Request $request)
+    {
+        //REQUEST DOES NOT HAVE TRADE_ID NOR CHARACTER_ID . VALIDATION FAILS
+        $offerData = $request->validate([
+            'trade_id' => 'required|exists:trades,id|integer',
+            'character_id' => "required|exists:characters,id|integer",
+        ]);
+
+        $character = Character::where('user_id', Auth::id())
+            ->findOrFail($offerData['character_id']);
+        $trade = Trade::findOrFail($offerData['trade_id']);
+        $tradeItem = $trade->item()->get();
+
+        $validItems = $character->items()->where('tier', $tradeItem->tier);
+
+        return view('offer.create', compact('validItems'));
+    }
```

Comment:
you can just do this
```suggestion
    public function create(Request $request)
    {
        $offerData = $request->validate([
            'trade_id' => 'required|exists:trades,id|integer',
        ]);
        
        $trade = Trade::with('item')->findOrFail($offerData['trade_id']);
        $validItems = Auth::user()->items()->where('rarity', $trade->item->rarity)
        
        // will be replaced when page component is created...
        return view('offer.create', compact('validItems', 'trade'));
    }
```

Reply:
@m-triassi is it tier or rarity that we're looking for?

Reply:
I modified this suggestion to accept a trade parameter instead. Adjusted route and tests are now passing